File: benchmarks/pandas/bench_index_isin.py
"""Benchmark: index_isin — pandas Index.isin() membership check on 100k-element Index"""
import json
import time
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

idx = pd.Index(range(ROWS))
lookup = list(range(0, ROWS, 100))

for _ in range(WARMUP):
    _ = idx.isin(lookup)

start = time.perf_counter()
for _ in range(ITERATIONS):
    _ = idx.isin(lookup)
total = (time.perf_counter() - start) * 1000

print(json.dumps({"function": "index_isin", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))

File: benchmarks/tsb/bench_index_isin.ts
/**
 * Benchmark: index_isin — Index.isin() membership check on 100k-element Index
 */
import { Index } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const idx = new Index(Array.from({ length: ROWS }, (_, i) => i));
const lookup = Array.from({ length: 1_000 }, (_, i) => i * 100);

for (let i = 0; i < WARMUP; i++) {
  idx.isin(lookup);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  idx.isin(lookup);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "index_isin",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
